packages/cashier-v2/src/lib/Transfer/Transfer.tsx

```diff
@@ -1,32 +1,53 @@
-import React, { useCallback, useEffect } from 'react';
-import { useActiveAccount, useCurrencyConfig, useTransferBetweenAccounts } from '@deriv/api-v2';
-import { Loader } from '@deriv-com/ui';
-import { TransferForm } from './components';
-import { TransferProvider } from './provider';
-
-const Transfer = () => {
-    return <TransferForm />;
+import React from 'react';
+import { useAccountLimits, useActiveAccount, useCurrencyConfig } from '@deriv/api-v2';
+import { Loader, Text } from '@deriv-com/ui';
+import { ErrorDialog } from '../../components';
+import { THooks } from '../../hooks/types';
+import { TransferForm, TransferReceipt } from './components';
+import { TransferProvider, useTransfer } from './provider';
+
+type TTransferModuleProps = {
+    accounts?: THooks.TransferAccounts;
 };
 
-const TransferModule = () => {
-    const { data: activeAccount } = useActiveAccount();
-    const { getConfig } = useCurrencyConfig();
-    const { data: transferAccounts, mutate } = useTransferBetweenAccounts();
-    const accounts = transferAccounts?.accounts;
+const Transfer = ({ isImported }: { isImported: boolean }) => {
+    const { setTransferReceipt, transferError, transferReceipt } = useTransfer();
+
+    if (transferReceipt) return <TransferReceipt {...transferReceipt} resetTransferReceipt={setTransferReceipt} />;
 
-    const requestForAccounts = useCallback(() => mutate({ accounts: 'all' }), [mutate]);
+    return (
+        <>
+            <TransferForm />
+            {/* isImported checks if the module is used as an import from the cashier-v2 library */}
+            {!isImported && (
+                <ErrorDialog isOpen={!!transferError}>
+                    <Text as='p' size='sm'>
+                        {transferError?.error.message}
+                    </Text>
+                </ErrorDialog>
+            )}
+        </>
+    );
+};
 
-    useEffect(() => {
-        requestForAccounts();
-    }, [requestForAccounts]);
+const TransferModule: React.FC<TTransferModuleProps> = ({ accounts }) => {
+    const { data: activeAccount } = useActiveAccount();
+    const { getConfig, isLoading: isCurrencyConfigLoading } = useCurrencyConfig();
+    const { data: accountLimits, refetch } = useAccountLimits();
 
-    const isLoading = !accounts || !activeAccount || !getConfig;
+    const isLoading = !activeAccount || !accountLimits || isCurrencyConfigLoading;
 
     if (isLoading) return <Loader />;
 
     return (
-        <TransferProvider accounts={accounts} activeAccount={activeAccount} getConfig={getConfig}>
-            <Transfer />
+        <TransferProvider
+            accountLimits={accountLimits}
+            accounts={accounts}
+            activeAccount={activeAccount}
+            getConfig={getConfig}
+            refetchAccountLimits={refetch}
+        >
+            <Transfer isImported={!accounts} />
```

Comment:
hmm, what does it mean, that "isImported" (in context of import from cashier-v2, so in context of technical import) is false when there are no accounts returned from api?